### ISSUE TITLE
Admin redirect loop

### ISSUE THREAD

**Issue by michalmikolajczyk** [bitnami/bitnami-docker-magento#123]
This is a bug very similar to the one described at #64 

I just installed Magento from the helm chart literally a couple of days back. Kubernetes & Istio. Frontend works fine, no problems. The admin pages are getting a redirect loop.

My findings so far:
```
mysql -hmanageable-gnat-mariadb --port 3306 -ubn_magento -pmypassword bitnami_magento -Bse "select * from core_config_data where path like 'web/%';"
```
returns
```
3       default 0       web/unsecure/base_url   http://magento.mydomain.cloud/      2019-11-13 14:04:20
4       default 0       web/secure/base_url     https://magento.mydomain.cloud/     2019-11-14 10:45:15
6       default 0       web/secure/use_in_frontend      1       2019-11-14 23:40:40
7       default 0       web/secure/use_in_adminhtml     1       2019-11-14 23:40:53
40      default 0       web/url/redirect_to_base        0       2019-11-14 10:06:20
41      default 0       web/cookie/cookie_domain        magento.mydomain.cloud      2019-11-13 14:38:58
```

the cache was cleaned, the app reacts to changes in those settings... e.g. when I change `use_in_adminhtml` to 0, and visit the page still via https, I start getting both 301 and 302 redirects, whereas with the setting set to 1 I only get 302 redirect loop.

**The same problem happens with or without https!** when the when the redirect on webserver is off and `use_in_adminhtml` is set to 0, the redirect just occurs on `http` protocol.

**Comment by andresbono** [comment]
@michalmikolajczyk, in order to reproduce your issue, can you share the values that you used to deploy the Magento chart? Did you do any modifications to the application?

**Comment by michalmikolajczyk** [comment]
I just deployed it with Helm without providing any values initially, then updated the values for Magento Host etc – using Helm upgrade, based on the logs.

**Comment by andresbono** [comment]
We couldn't reproduce your issue. We will need more details about the cluster where you are deploying the helm chart and the specific version of the chart that you are launching. Also, how did you configure the domain name on Magento and the HTTPS configuration? Can you share the specific helm commands that you used to deploy the solution?

**Comment by michalmikolajczyk** [comment]
Ok I managed to resolve the problem.

I installed magento like this: `helm install --set magentoHost=magento.mydomain.cloud bitnami/magento`

The problem was not in the installation, but in our network configuration. Even though Magento was accessed using https, the TLS termination occurs at the load balancer (or in our case, at the Istio Gateway).

That means, Magento receives the request from the Gateway in pure http, but requested url was with https. To avoid the redirect loop, those were the settings I used:

```
web/url/redirect_to_base 0
web/secure/use_in_frontend 0
web/secure/use_in_adminhtml 0
web/cookie/cookie_domain magento.mydomain.cloud
web/secure/base_url https://magento.mydomain.cloud
web/unsecure/base_url https://magento.mydomain.cloud
```

**NOTE** the **https** by unsecure base url and 0 values set for `web/secure/*` which sets it so Magento does not handle TLS termination, but will only handle `https` requests.
